android/java/org/chromium/chrome/browser/brave_news/BraveNewsUtils.java

```diff
@@ -0,0 +1,128 @@
+import org.chromium.brave_news.mojom.Publisher;
+import org.chromium.chrome.browser.brave_news.models.FeedItemCard;
+import org.chromium.chrome.browser.brave_news.models.FeedItemsCard;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Iterator;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class BraveNewsUtils {
+    private Context mContext;
+
+    public BraveNewsUtils(Context context) {
+        mContext = context;
+    }
+
+    private Publisher getPublisher(){
+        Publisher publisher = null;
+
+        return publisher;
+
+    }
+
+    public String getPromotionIdItem(FeedItemsCard items) {
+        String creativeInstanceId = "null";
+        if (items.getFeedItems() != null){            
+            for (FeedItemCard itemCard : items.getFeedItems()) {
+                FeedItem item = itemCard.getFeedItem();
+                FeedItemMetadata itemMetaData = new FeedItemMetadata();
+                switch (item.which()) {
+                    case FeedItem.Tag.PromotedArticle:
+                        PromotedArticle promotedArticle = item.getPromotedArticle();
+                        FeedItemMetadata promotedArticleData = promotedArticle.data;
+                        creativeInstanceId = promotedArticle.creativeInstanceId;
+                        itemMetaData = promotedArticle.data;
+                        break;
+                }
+            }
+        }
+
+        return creativeInstanceId;
+    }
+
+    // method for logging news object
+    public static void logFeedItem(FeedItemsCard items, String id){
+        if (items != null) {
+            if (items.getCardType() == CardType.DISPLAY_AD){
+                DisplayAd displayAd = items.getDisplayAd();
+                if (displayAd != null){                    
+                    Log.d("bn", id + " DISPLAY_AD title: " + displayAd.title);
+                }
+            } else {
+                if (items.getFeedItems() != null){       
+                    int index = 0;          
+                    for (FeedItemCard itemCard : items.getFeedItems()){
+                        if (index > 50){
+                            return;
+                        }
+                        FeedItem item = itemCard.getFeedItem();
+
+                        FeedItemMetadata itemMetaData = new FeedItemMetadata();
+                        switch(item.which()){
+                            case FeedItem.Tag.Article:
+                                
+                                Article article = item.getArticle();
+                                FeedItemMetadata articleData = article.data;
+                                itemMetaData = article.data;
+                                Log.d("bn", id + " articleData: " + articleData.title);
+                                // Log.d("bn", id + " articleData categoryName: " + articleData.categoryName);
```

Comment:
should these commented lines be here still?

Reply:
it's a method for logging, it shouldn't be called unless that's needed. not sure if it should be kept, maybe at least until it passes QA. i removed the Logs for now